### ISSUE TITLE
Missing (all) stencil formats

### ISSUE THREAD

**Issue by kopaka1822** [shader-slang/slang#2884]
I noticed that all formats that provide a stencil have been removed.
I saw the comments, that D24_S8 is only supported on NVIDIA devices.
However, there is no reason to exclude D32_S8?

I think these formats are still relevant since the provide a good speedup for selective per-pixel operations when compared to using the discard intrinsic in the pixel shader.

**Comment by kopaka1822** [comment]
I also noticed that the stencilRef value appears to be unused in DirectX12, since this is not part of the PipelineStateObject, but instead needs to be set via the command list (ID3D12GraphicsCommandList::OMSetStencilRef)

**Comment by csyonghe** [comment]
You are right. The formats are missing mainly because we haven't run into a use case that needs it.
Your format PR looks great. Thank you for your contribution!

**Comment by natduca** [comment]
I think this is patched, closing. If mistake feel free to reopen.
